client/src/main/java/org/apache/nemo/client/ClientEndpoint.java

```diff
@@ -90,7 +90,12 @@ private boolean waitUntilConnected(final long timeout,
     try {
       if (driverEndpoint.get() == null) {
         // If the driver endpoint is not connected, wait.
-        return driverConnected.await(timeout, unit);
+        if (timeout < 1) {
```

Comment:
timeout == 0 ?

Reply:
or why don't you use `timeout == Long.MAX_VALUE `? and call `waitUntilConnected(Long.MAX_VALUE, TimeUnit.SECONDS)`?

Reply:
Alright, the max value of long `9,223,372,036,854,775,807` seems pretty much large enough to cover the timeout.